reports/generator/generator_test.go

```diff
@@ -0,0 +1,306 @@
+	}
+	fileNoDataReportRow = []string{"file_id_1", "version_1", "patient_1", "2018-07-27T13:55:59.123Z", "2018-07-29T13:55:59.123Z", "", "", ""}
+
+	fileAllData = reports.File{
+		FileID:    "file_id_1",
+		Version:   "version_1",
+		PatientID: "patient_1",
+		CreatedAt: time1,
+		UpdatedAt: time2,
+		Data:      "{\"/userID\": \"ID\", \"/userName\": \"username\", \"/arrayLevel1:0/arrayLevel2:0/nestedArrayItemID\": \"ID0:0\", \"/arrayLevel1:0/arrayLevel2:0/nestedArrayItemName\": \"Name0:0\", \"/arrayLevel1:0/arrayLevel2:1/nestedArrayItemID\": \"ID0:1\", \"/arrayLevel1:0/arrayLevel2:1/nestedArrayItemName\": \"Name0:1\", \"/arrayLevel1:0/additionalData\": \"Data0\", \"/arrayLevel1:1/arrayLevel2:0/nestedArrayItemID\": \"ID1:0\", \"/arrayLevel1:1/arrayLevel2:0/nestedArrayItemName\": \"Name1:0\", \"/arrayLevel1:1/arrayLevel2:1/nestedArrayItemID\": \"ID1:1\", \"/arrayLevel1:1/arrayLevel2:1/nestedArrayItemName\": \"Name1:1\", \"/arrayLevel1:1/additionalData\": \"Data1\", \"/arrayLevel1:2/arrayLevel2:0/nestedArrayItemID\": \"ID2:0\", \"/arrayLevel1:2/arrayLevel2:0/nestedArrayItemName\": \"Name2:0\", \"/arrayLevel1:2/arrayLevel2:1/nestedArrayItemID\": \"ID2:1\", \"/arrayLevel1:2/arrayLevel2:1/nestedArrayItemName\": \"Name2:1\", \"/arrayLevel1:2/additionalData\": \"Data2\"}",
+	}
+	fileAllDataReportRow = []string{"file_id_1", "version_1", "patient_1", "2018-07-27T13:55:59.123Z", "2018-07-29T13:55:59.123Z", "ID username", "ID0:0/Name0:0, ID0:1/Name0:1 - Data0", "ID1:0/Name1:0, ID1:1/Name1:1 - Data1, ID2:0/Name2:0, ID2:1/Name2:1 - Data2"}
+
+	fileMissingData = reports.File{
+		FileID:    "file_id_1",
+		Version:   "version_1",
+		PatientID: "patient_1",
+		CreatedAt: time1,
+		UpdatedAt: time2,
+		Data:      "{\"/userID\": \"ID\",  \"/arrayLevel1:0/arrayLevel2:0/nestedArrayItemID\": \"ID0:0\", \"/arrayLevel1:0/arrayLevel2:0/nestedArrayItemName\": \"Name0:0\", \"/arrayLevel1:0/arrayLevel2:1/nestedArrayItemID\": \"ID0:1\", \"/arrayLevel1:0/additionalData\": \"Data0\"}",
+	}
+	fileMissingDataReportRow = []string{"file_id_1", "version_1", "patient_1", "2018-07-27T13:55:59.123Z", "2018-07-29T13:55:59.123Z", "ID", "ID0:0/Name0:0, ID0:1/ - Data0", ""}
+
+	invalidJSONDataFile = reports.File{
+		FileID:    "file_id_1",
+		Version:   "version_1",
+		PatientID: "patient_1",
+		CreatedAt: time1,
+		UpdatedAt: time2,
+		Data:      "{this is invalid json}",
+	}
+)
+
+func TestGenerate(t *testing.T) {
+	testCases := []struct {
+		description    string
+		reportSpec     ReportSpec
+		createdAtStart *strfmt.DateTime
+		createdAtEnd   *strfmt.DateTime
+		calls          func(*storageMock.MockStorage, *generatorMock.MockReportWriter)
+		errorExpected  bool
+		exactError     error
+	}{
+		{
+			"Only empty file returned",
+			testSpec,
+			nil,
+			nil,
+			func(storage *storageMock.MockStorage, reportWriter *generatorMock.MockReportWriter) {
+				gomock.InOrder(
+					storage.EXPECT().Find("", map[string]string{"/category": "openehr::111|test|"}, nil, nil).Return(
```

Comment:
invalid operation: storage (variable of type *invalid type) has no field or method EXPECT